From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Tue, 26 Apr 2016 14:53:59 +0800
Subject: mmc: dw_mmc-rockchip: add MMC_CAP_CMD23 capabilities
Git-commit: d57fe83e0744cdc88236226980c613435ad5c10c
Patch-mainline: v4.7-rc1
References: fate#322003

Add MMC_CAP_CMD23 for dw_mmc-rockchip, otherwise
failing to create rpmb partition. With it, we can
get rpmb successfully:

mmc1: new HS200 MMC card at address 0001
mmcblk0: mmc1:0001 DS2016 14.7 GiB
mmcblk0boot0: mmc1:0001 DS2016 partition 1 4.00 MiB
mmcblk0boot1: mmc1:0001 DS2016 partition 2 4.00 MiB
mmcblk0rpmb: mmc1:0001 DS2016 partition 3 4.00 MiB

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
Signed-off-by: Jaehoon Chung <jh80.chung@samsung.com>
Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/mmc/host/dw_mmc-rockchip.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/mmc/host/dw_mmc-rockchip.c
+++ b/drivers/mmc/host/dw_mmc-rockchip.c
@@ -238,10 +238,10 @@ static int dw_mci_rockchip_init(struct d
 
 /* Common capabilities of RK3288 SoC */
 static unsigned long dw_mci_rk3288_dwmmc_caps[4] = {
-	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* emmc */
-	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdmmc */
-	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdio0 */
-	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdio1 */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE | MMC_CAP_CMD23, /* emmc */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE | MMC_CAP_CMD23, /* sdmmc */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE | MMC_CAP_CMD23, /* sdio0 */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE | MMC_CAP_CMD23, /* sdio1 */
 };
 static const struct dw_mci_drv_data rk2928_drv_data = {
 	.prepare_command        = dw_mci_rockchip_prepare_command,
